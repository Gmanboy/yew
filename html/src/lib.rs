use proc_macro_hack::proc_macro_hack;

/// Generate html tree
#[proc_macro_hack]
pub use yew_html_impl::html;
