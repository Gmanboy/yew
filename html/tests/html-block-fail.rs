use yew_html::html;

struct NotHtmlTree {}

fn not_tree() -> NotHtmlTree {
    NotHtmlTree {}
}

fn main() {
    html! {
        { not_tree() }
    };

    html! {
        {
            let not_a_tree = not_tree();
            not_a_tree
        }
    };

    html! {
        <>
            { (0..3).map(|_| not_tree()) }
        </>
    };
}
