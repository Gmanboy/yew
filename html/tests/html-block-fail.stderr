error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/html-block-fail.rs:10:5
   |
10 | /     html! {
11 | |         <div>{ not_tree() }</div>
12 | |     }
   | |_____^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::string::ToString` for `()`
   = note: required because of the requirements on the impl of `std::convert::From<()>` for `yew::virtual_dom::vnode::VNode<_>`
   = note: required by `std::convert::From::from`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/html-block-fail.rs:18:5
   |
18 | /     html! {
19 | |         <>{ for (0..3).map(|_| not_tree()) }</>
20 | |     }
   | |_____^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::string::ToString` for `()`
   = note: required because of the requirements on the impl of `std::convert::From<()>` for `yew::virtual_dom::vnode::VNode<_>`
   = note: required by `std::convert::From::from`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> $DIR/html-block-fail.rs:3:1
  |
3 | / test_html! { |t1|
4 | |     { () }
5 | | }
  | |_^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required because of the requirements on the impl of `std::string::ToString` for `()`
  = note: required because of the requirements on the impl of `std::convert::From<()>` for `yew::virtual_dom::vnode::VNode<_>`
  = note: required by `std::convert::From::from`
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
